src/ophyd_async/core/async_status.py

```diff
@@ -94,3 +86,59 @@ def __repr__(self) -> str:
         return f"<{type(self).__name__}, task: {self.task.get_coro()}, {status}>"
 
     __str__ = __repr__
+
+
+class AsyncStatus(AsyncStatusBase):
+    @classmethod
+    def wrap(cls: Type[AS], f: Callable[P, Awaitable]) -> Callable[P, AS]:
+        @functools.wraps(f)
+        def wrap_f(*args: P.args, **kwargs: P.kwargs) -> AS:
+            return cls(f(*args, **kwargs))
+
+        # type is actually functools._Wrapped[P, Awaitable, P, AS]
+        # but functools._Wrapped is not necessarily available
+        return cast(Callable[P, AS], wrap_f)
+
+
+class WatchableAsyncStatus(AsyncStatusBase, Generic[T]):
+    """Convert AsyncIterator of WatcherUpdates to bluesky Status interface."""
+
+    def __init__(
+        self,
+        iterator_or_awaitable: Awaitable | AsyncIterator[WatcherUpdate[T]],
+        watchers: list[Watcher] = [],
+    ):
+        self._watchers: list[Watcher] = watchers
+        self._start = time.monotonic()
+        self._last_update: WatcherUpdate[T] | None = None
+        awaitable = (
+            iterator_or_awaitable
+            if isinstance(iterator_or_awaitable, Awaitable)
+            else self._notify_watchers_from(iterator_or_awaitable)
+        )
+        super().__init__(awaitable)
+
+    async def _notify_watchers_from(self, iterator: AsyncIterator[WatcherUpdate[T]]):
+        async for self._last_update in iterator:
+            for watcher in self._watchers:
+                self._update_watcher(watcher, self._last_update)
+
+    def _update_watcher(self, watcher: Watcher, update: WatcherUpdate[T]):
+        watcher(replace(update, time_elapsed_s=time.monotonic() - self._start))
```

Comment:
Given https://blueskyproject.io/bluesky/hardware.html#watch we need parameters to be passed down individually, so it would be nice if we could make the dataclass args the same as the signature so we could do:
```suggestion
        watcher(**update.dict())
```